discojs/src/client/event_connection.ts

```diff
@@ -79,8 +79,11 @@ export class PeerConnection extends EventEmitter<{ [K in type]: NarrowMessage<K>
     this.peer.send(msgpack.encode(msg))
   }
 
-  async disconnect (): Promise<void> {
-    await this.peer.destroy()
+  async disconnect(): Promise<void> {
+    await Promise.race([
+      this.peer.destroy(),
+      new Promise((res, _) => setTimeout(() => res('PeerConnection disconnection timed out'), 3000))
+    ])
```

Comment:
that's not our purview. if it's taking too long, we need to work on the wrtc stack itself. either with a documented upstream issue or even a PR.

```suggestion
    await this.peer.destroy()
```